packages/bsky/src/auth-verifier.ts

```diff
@@ -440,5 +451,54 @@ export const buildBasicAuth = (username: string, password: string): string => {
 const keyEncoder = new KeyEncoder('secp256k1')
 export const createPublicKeyObject = (publicKeyHex: string): KeyObject => {
   const key = keyEncoder.encodePublic(publicKeyHex, 'raw', 'pem')
-  return createPublicKey({ format: 'pem', key })
+  return crypto.createPublicKey({ format: 'pem', key })
+}
+
+export const verifySignatureWithKey: VerifySignatureWithKeyFn = async (
+  didKey: string,
+  msgBytes: Uint8Array,
+  sigBytes: Uint8Array,
+  alg: string,
+) => {
+  if (alg === SECP256K1_JWT_ALG) {
+    const parsed = parseDidKey(didKey)
+    if (alg !== parsed.jwtAlg) {
+      throw new Error(`Expected key alg ${alg}, got ${parsed.jwtAlg}`)
+    }
+
+    const prefixedBytes = extractPrefixedBytes(extractMultikey(didKey))
+    if (!hasPrefix(prefixedBytes, SECP256K1_DID_PREFIX)) {
+      throw new Error(`Not a secp256k1 did:key: ${didKey}`)
+    }
+    const keyBytes = prefixedBytes.slice(SECP256K1_DID_PREFIX.length)
+
+    const verifySig = (
```

Comment:
actually same goes for this whole function. I don't think we're using a closure here, so we can probably just pull this out & only define the function once rather than once per call